### PR TITLE
Refactor TestDelegatingConnection and TestConnectionPool

### PR DESCRIPTION
**Pull request by rajucomp**
Thanks for your contribution to [Apache Commons](https://commons.apache.org/)! Your help is appreciated!

Before you push a pull request, review this list:

- [x] Read the [contribution guidelines](CONTRIBUTING.md) for this project.
- [x] Run a successful build using the default [Maven](https://maven.apache.org/) goal with `mvn`; that's `mvn` on the command line by itself.
- [x] Write unit tests that match behavioral changes, where the tests fail if the changes to the runtime are not applied. This may not always be possible but is a best-practice.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Each commit in the pull request should have a meaningful subject line and body. Note that commits might be squashed by a maintainer on merge.

### CODE REVIEW

**Comment by rajucomp** [comment]
@garydgregory This should be worth your time.

File: src/test/java/org/apache/commons/dbcp2/TestDelegatingConnection.java [hunk 1]
```diff
@@ -134,12 +134,7 @@ public void testAutoCommitCaching() throws SQLException {
     public void testCheckOpen() throws Exception {
         delegatingConnection.checkOpen();
         delegatingConnection.close();
-        try {
-            delegatingConnection.checkOpen();
-            fail("Expecting SQLException");
-        } catch (final SQLException ex) {
-            // expected
-        }
+        assertThrows(SQLException.class, () -> delegatingConnection.checkOpen(), "Expecting SQLException");
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/dbcp2/TestDelegatingConnection.java:137]: Good one 👍  but this should be applied in one PR for all uses of this pattern. Rebase on git master to find the whole code base using `assertThrows()`.

**Review by garydgregory — CHANGES_REQUESTED** [review]
-1: It looks like the test semantics have changed because operations and assertions now occur in a different order, the tests are no longer testing the same thing.

**Comment by rajucomp** [comment]
> -1: It looks like the test semantics have changed because operations and assertions now occur in a different order, the tests are no longer testing the same thing.

Does this mean the tests are no longer valid and needs to be updated ? I am happy to do those. Will need some guidance on how to approach those.

**Comment by rajucomp** [comment]
Okay. I get it. My bad. Let me fix the tests to preserve the operations and the assertions order.

**Comment by garydgregory** [comment]
@rajucomp 
I don't know and TBH I don't think this is a good use of maintainers' time. If you're looking for something to do, you can look at bug reports in Jira.

**Comment by rajucomp** [comment]
That's a fair callout. I will close this PR.

**Pull request closed without merging**